news/_posts/2020-02-11-phpunit-security-issue-post-analysis.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+title:  "Security issue with PHPUnit, post-incident analysis"
+subtitle: "The mysterious development dependencies!"
+date:   2020-02-11 08:00:00
+authors: [ PierreRambaud, PrestaShop ]
+icon: icon-chat
+image: /assets/images/theme/meta-logo-build.png
+tags: [maintenance, patch, modules]
+---
+
+Early January, we encountered a [security issue with PHPUnit](https://build.prestashop.com/news/critical-security-vulnerability-in-prestashop-modules/) in some modules that allowed attackers to perform arbitrary code execution without authorization through the PHPUnit dependency. This vulnerability was discovered through a merchant whose shops were compromised. 
+
+## Background
+
+At PrestaShop, there is an automatic system on the Addons Marketplace which gets the module from the [PrestaShop-modules](https://github.com/PrestaShop/PrestaShop-modules) repository and uses Composer to install dependencies to be distributed by the API. During this step, by mistake, development dependencies have been built inside.
+ 
+This was a code error, fixed after a short time, but the bad archives were already downloaded and unfortunately some archives which contained PHPUnit were not properly cleaned up.
```

Comment:
```suggestion
This was a code error which was fixed after a short time, but the badly built archives had already been distributed to active shops. At that time, this was considered not to be an issue, since the number of modules including development dependencies was fairly small, and we weren't aware of any vulnerability involving said dependencies.
```